pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -117,7 +122,12 @@ func (s *Reconciler) CreateMachine(ctx context.Context) error {
 		return fmt.Errorf("failed to create nic %s for machine %s: %w", nicName, s.scope.Machine.Name, err)
 	}
 
-	if err := s.createVirtualMachine(ctx, nicName); err != nil {
+	asName, err := s.createAvailabilitySet()
```

Comment:
would appreciate comment there which describes behavior of `createAvailabilitySet`. Now its not obvious that no availability set would be created and empty string will be returned if zones found.